operatorcourier/flatten.py

```diff
@@ -0,0 +1,151 @@
+import logging
+import os
+from yaml import safe_load, MarkedYAMLError
+import semver
+import ntpath
+import operatorcourier.identify as identify
+import operatorcourier.errors as errors
+
+logger = logging.getLogger(__name__)
+
+
+def get_flattened_files_info(source_dir: str) -> [(str, str)]:
+    """
+    :param source_dir: Path of the directory containing different versions
+    of operator bundles (CRD, CSV, package) in separate version directories
+    :return: A list of tuples where in each tuple, the first element is
+    the source file path to be copied to the flattened directory, and the second is
+    the new file name to be used in the file copy. The function returns an empty list if
+    the source_dir is already flat
+    """
+
+    # extract package file and version folders from source_dir
+    _, folder_names, file_names = next(os.walk(source_dir))
+    if len(folder_names) == 0:
+        logger.info('The source directory is already flat. Exiting the program...')
```

Comment:
Don't exit anything here, this is part of API, I'm sure I don't want SysOps to see this message in service logs. Drop `Exiting the program...` part please.